invenio_communities/assets/semantic-ui/js/invenio_communities/members/invitations/InvitationResultItem.js

```diff
@@ -6,146 +6,84 @@
 
-export const InvitationResultItem = ({
-  invitation,
-  onRoleChange,
-  isLoading,
-  error,
-  success,
-  onSuccessTimeOut,
-  onErrorClose,
-  roles,
-  onView,
-  onReInvite,
-  onCancel,
-}) => {
-  const { receiver, status, expires_at, links } = invitation;
+export class InvitationResultItem extends Component {
+  constructor(props) {
+    super(props);
+    const { result } = this.props;
+    this.state = { invitation: result };
+    this.invitationsApi = new CommunityInvitationsApi(result.links);
+  }
 
-  return (
-    <Table.Row className="community-member-item">
-      <Table.Cell>
-        <Grid textAlign="left" verticalAlign="middle">
-          <Grid.Column width={4}>
-            <Image src={"/abc"} avatar circular />
-          </Grid.Column>
-          <Grid.Column width={12}>
-            <List>
-              <List.Item as="a">{receiver.name || receiver.user}</List.Item>
-            </List>
-          </Grid.Column>
-        </Grid>
-      </Table.Cell>
-      <Table.Cell>{status}</Table.Cell>
-      <Table.Cell>
-        <Grid>
-          <Grid.Row>
-            <Grid.Column width={13} verticalAlign="middle">
-              {formattedTime(expires_at)}
-            </Grid.Column>
-            <Grid.Column width={3} textAlign="right">
-              {isLoading && <Loader active inline="centered" size="small" />}
-              <SuccessIcon
-                success={success}
-                timeOutDelay={4000}
-                onTimeOut={onSuccessTimeOut}
-              />
-              <ErrorPopup
-                onClose={onErrorClose}
-                error={error}
-                trigger={<Icon name="exclamation circle" className="error" />}
-              />
-            </Grid.Column>
-          </Grid.Row>
-        </Grid>
-      </Table.Cell>
-      <Table.Cell>
-      <Dropdown
-        options={dropdownOptionsGenerator(roles)}
-        selection
-        fluid
-        // TODO - replace with role added to invitation payload
-        value="curator"
-        openOnFocus={false}
-        selectOnBlur={false}
-        onChange={(e, { value: role }) => {
-            onRoleChange(role);
-          }}
-      />
-      </Table.Cell>
-      <Table.Cell>
-        <Container fluid textAlign="right">
-          <ActionButtons
-            actions={links.actions}
-            onCancel={onCancel}
-            onReInvite={onReInvite}
-            onView={onView}
-            status={status}
+  updateInvitation = (data) => {
+    const { invitation } = this.state;
+    this.setState({ invitation: { ...invitation, ...data } });
+  };
+
+  render() {
+    const {
+      config: { roles },
+    } = this.props;
+    const {
+      invitation,
+      invitation: { receiver, status, expires_at, role },
+    } = this.state;
+
+    return (
+      <Table.Row className="community-member-item">
+        <Table.Cell>
+          <Image src={"/abc"} avatar circular className="rel-mr-1" />
```

Comment:
fake image src for now

Reply:
is it documented somewhere that it should be fixed? An issue?